pkg/controllers/management/clusteroperator/utils.go

```diff
@@ -251,3 +264,39 @@ func (e *OperatorController) RecordCAAndAPIEndpoint(cluster *mgmtv3.Cluster) (*m
 
 	return currentCluster, err
 }
+
+func GenerateSAToken(restConfig *rest.Config) (string, error) {
+	clientSet, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return "", fmt.Errorf("error creating clientset: %v", err)
+	}
+
+	return util.GenerateServiceAccountToken(clientSet)
+}
+
+func addAdditionalCA(secretsCache wranglerv1.SecretCache, caCert string) (string, error) {
+	additionalCA, err := getAdditionalCA(secretsCache)
+	if err != nil || additionalCA == nil {
+		return caCert, err
+	}
+
+	caBytes, err := base64.StdEncoding.DecodeString(caCert)
+	if err != nil {
+		return "", err
+	}
+
+	return base64.StdEncoding.EncodeToString(append(caBytes, additionalCA...)), nil
+}
+
+func getAdditionalCA(secretsCache wranglerv1.SecretCache) ([]byte, error) {
+	secret, err := secretsCache.Get(namespace.System, "tls-ca-additional")
+	if err != nil && !errors.IsNotFound(err) {
+		return nil, err
+	}
+
+	if secret == nil {
+		return nil, nil
+	}
+
+	return secret.Data["ca-additional.pem"], nil
```

Comment:
If they mess up secret and don't have this key, will this return an empty string or nil ? Might want to do this check on line 297.

Reply:
It returns `nil`. So it's possible this function returns `nil, nil` in a couple different places and the caller would have to handle these possibilities. There are two places where this function is used, and they both handle these possibilities.